src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,206 @@
+    private final Supplier<Double> toleranceRight = rightPIDNamespace.addConstantDouble("tolerance", 0);
+    private final PIDSettings rightPIDSettings;
+
+    private final Namespace anglePIDNamespace = namespace.addChild("angle pid");
+    private final Supplier<Double> kPAngle = anglePIDNamespace.addConstantDouble("kP", 0);
+    private final Supplier<Double> kIAngle = anglePIDNamespace.addConstantDouble("kI", 0);
+    private final Supplier<Double> kDAngle = anglePIDNamespace.addConstantDouble("kD", 0);
+    private final Supplier<Double> waitTimeAngle = anglePIDNamespace.addConstantDouble("wait time", 0);
+    private final Supplier<Double> toleranceAngle = anglePIDNamespace.addConstantDouble("tolerance", 0);
+    private final PIDSettings anglePIDSettings;
+
+    private final Namespace feedForwardNamespace = namespace.addChild("feed forward");
+    private final Supplier<Double> kS = feedForwardNamespace.addConstantDouble("kS", 0);
+    private final Supplier<Double> kV = feedForwardNamespace.addConstantDouble("kV", 0);
+    private final Supplier<Double> kA = feedForwardNamespace.addConstantDouble("kA", 0);
+    private final FeedForwardSettings feedForwardSettings;
+
+    private final Namespace trapezoidSettingsNamespace = namespace.addChild("trapezoid profile settings");
+    private final Supplier<Double> trapezoidVelocity = trapezoidSettingsNamespace.addConstantDouble("velocity", 0);
+    private final Supplier<Double> trapezoidAcceleration = trapezoidSettingsNamespace.addConstantDouble
+            ("acceleration", 0);
+    private final TrapezoidProfileSettings trapezoidProfileSettings;
+
+    public static Drivetrain getInstance() {
+        if (instance == null) {
+            instance = new Drivetrain(
+                    "drivetrain",
+                    new CANSparkMax(RobotMap.CAN.DRIVETRAIN_LEFT_SPARKMAX_1,
+                            CANSparkMaxLowLevel.MotorType.kBrushless),
+                    new CANSparkMax(RobotMap.CAN.DRIVETRAIN_LEFT_SPARKMAX_2,
+                            CANSparkMaxLowLevel.MotorType.kBrushless),
+                    new CANSparkMax(RobotMap.CAN.DRIVETRAIN_RIGHT_SPARKMAX_1,
+                            CANSparkMaxLowLevel.MotorType.kBrushless),
+                    new CANSparkMax(RobotMap.CAN.DRIVETRAIN_RIGHT_SPARKMAX_2,
+                            CANSparkMaxLowLevel.MotorType.kBrushless));
+        }
+        return instance;
+    }
+
+    private Drivetrain(String namespaceName, CANSparkMax leftMaster, CANSparkMax leftSlave,
+                       CANSparkMax rightMaster, CANSparkMax rightSlave) {
+        super(
+                namespaceName, leftMaster, leftSlave, rightMaster, rightSlave);
+        this.gyro = new AHRS();
+        this.leftEncoder = leftMaster.getEncoder();
+        this.rightEncoder = rightMaster.getEncoder();
+        this.trapezoidProfileSettings = new TrapezoidProfileSettings(trapezoidVelocity, trapezoidAcceleration);
```

Comment:
Move to fields to be initialized in this order:
leftPID
rightPID
anglePID
feedForwardSettings
trapezoidProfileSettings